src/core/prometheus.cc

```diff
@@ -531,9 +532,11 @@ void write_summary(std::stringstream& s, const config& ctx, const sstring& name,
  */
 class metric_aggregate_by_labels {
     std::vector<std::string> _labels_to_aggregate_by;
+    metrics::aggregate_function_type _aggregate_function;
     std::unordered_map<std::map<sstring, sstring>, seastar::metrics::impl::metric_value> _values;
+    std::unordered_map<std::map<sstring, sstring>, size_t> _count;
 public:
-    metric_aggregate_by_labels(std::vector<std::string> labels) : _labels_to_aggregate_by(std::move(labels)) {
+    metric_aggregate_by_labels(std::vector<std::string> labels, metrics::aggregate_function_type agg_function) : _labels_to_aggregate_by(std::move(labels)), _aggregate_function(agg_function), _count(1) {
```

Comment:
why would you want to specify the bucket size explicitly? i think the default bucket size of `unordered_map` of libstdc++ is indeed 1. see https://github.com/gcc-mirror/gcc/blob/59b4a555c3f1c3dba376da1c4886a9ea18ad208d/libstdc%2B%2B-v3/include/bits/hashtable.h#L388